### PR TITLE
Added canRotate and aspectRatio attribute

### PR DESCRIPTION
**Pull request by demonskp**
Two attributes have been added to control whether rotation is allowed, and shapes that support a fixed aspect ratio

### CODE REVIEW

File: example/lib/resultScreen.dart [hunk 1]
```diff
@@ -10,6 +10,7 @@ class ResultScreen extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+
```
Comment (vanlooverenkoen) [inline example/lib/resultScreen.dart:13]: formatting

File: pubspec.lock [hunk 2]
```diff
@@ -5,49 +5,49 @@ packages:
     dependency: transitive
     description:
       name: async
-      url: "https://pub.dartlang.org"
+      url: "https://pub.flutter-io.cn"
```
Comment (vanlooverenkoen) [inline pubspec.lock:8]: revert this back to dartlang.org

File: example/pubspec.lock [hunk 3]
```diff
@@ -5,63 +5,63 @@ packages:
     dependency: transitive
     description:
       name: async
-      url: "https://pub.dartlang.org"
+      url: "https://pub.flutter-io.cn"
```
Comment (vanlooverenkoen) [inline example/pubspec.lock:8]: revert this back to dartlang.org

File: lib/src/widgets/custom_image_crop_widget.dart [hunk 4]
```diff
@@ -159,9 +165,9 @@ class _CustomImageCropState extends State<CustomImageCrop> with CustomImageCropL
 
   void onScaleUpdate(ScaleEvent event) {
     if (dataTransitionStart != null) {
-      addTransition(dataTransitionStart! - CropImageData(scale: event.scale, angle: event.rotationAngle));
+      addTransition(dataTransitionStart! - CropImageData(scale: event.scale, angle: widget.canRotate?event.rotationAngle:0));
     }
-    dataTransitionStart = CropImageData(scale: event.scale, angle: event.rotationAngle);
+    dataTransitionStart = CropImageData(scale: event.scale, angle: widget.canRotate?event.rotationAngle:0);
```
Comment (vanlooverenkoen) [inline lib/src/widgets/custom_image_crop_widget.dart:170]: formatting

File: lib/src/widgets/custom_image_crop_widget.dart [hunk 5]
```diff
@@ -172,14 +178,14 @@ class _CustomImageCropState extends State<CustomImageCrop> with CustomImageCropL
     addTransition(CropImageData(x: event.delta.dx, y: event.delta.dy));
   }
 
-  Path _getPath(double cropWidth, double width, double height) {
+  Path _getPath(double cropWidth,double cropHeight, double width, double height) {
```
Comment (vanlooverenkoen) [inline lib/src/widgets/custom_image_crop_widget.dart:181]: formatting

File: lib/src/widgets/custom_image_crop_widget.dart [hunk 6]
```diff
@@ -159,9 +165,9 @@ class _CustomImageCropState extends State<CustomImageCrop> with CustomImageCropL
 
   void onScaleUpdate(ScaleEvent event) {
     if (dataTransitionStart != null) {
-      addTransition(dataTransitionStart! - CropImageData(scale: event.scale, angle: event.rotationAngle));
+      addTransition(dataTransitionStart! - CropImageData(scale: event.scale, angle: widget.canRotate?event.rotationAngle:0));
```
Comment (vanlooverenkoen) [inline lib/src/widgets/custom_image_crop_widget.dart:168]: formatting

File: lib/src/widgets/custom_image_crop_widget.dart [hunk 7]
```diff
@@ -188,45 +194,69 @@ class _CustomImageCropState extends State<CustomImageCrop> with CustomImageCropL
...
+      if(width<height){
+        cropWidth = cropWidth * widget.cropPercentage;
+        cropHeight = cropWidth / widget.aspectRatio;
+      } else {
+        cropHeight = cropHeight * widget.cropPercentage;
+        cropWidth = cropHeight * widget.aspectRatio;
+      }
+    }
+
+    return {'cropWidth':cropWidth,'cropHeight':cropHeight};
```
Comment (vanlooverenkoen) [inline lib/src/widgets/custom_image_crop_widget.dart:222]: formatting

**Review by vanlooverenkoen — CHANGES_REQUESTED** [review]

**Comment by demonskp** [comment]
Reformat

File: lib/src/widgets/custom_image_crop_widget.dart [hunk 8]
```diff
@@ -188,45 +194,69 @@ class _CustomImageCropState extends State<CustomImageCrop> with CustomImageCropL
...
+  Map<String, double> _getCropSize(double width, double height){
+    double cropWidth = width;
+    double cropHeight = height;
+    
+    if(widget.shape == CustomCropShape.Circle){
+      cropWidth = min(width, height) *  widget.cropPercentage;
+      cropHeight = cropWidth;
+    }
+
+    if(widget.shape == CustomCropShape.Square){
```
Comment (ikbendewilliam) [inline lib/src/widgets/custom_image_crop_widget.dart:212]: I prefer a switch case, this way the linter will notify us if we missed a case (by adding new shapes for example)
Alternatively use an else if here so we don't do redundant checks

File: lib/src/widgets/custom_image_crop_widget.dart [hunk 9]
```diff
@@ -188,45 +194,69 @@ class _CustomImageCropState extends State<CustomImageCrop> with CustomImageCropL
...
+    final cropHeight = sizeMap['cropHeight'] ?? 0;
     final pictureRecorder = ui.PictureRecorder();
     final canvas = Canvas(pictureRecorder);
     final defaultScale = min(imageAsUIImage!.width, imageAsUIImage!.height) / cropWidth;
     final scale = data.scale * defaultScale;
-    final clipPath = Path.from(_getPath(cropWidth, cropWidth, cropWidth));
+    final clipPath = Path.from(_getPath(cropWidth, cropHeight, cropWidth, cropHeight));
     final matrix4Image = Matrix4.diagonal3(vector_math.Vector3(1, 1, 0))
-      ..translate(data.x + cropWidth / 2, data.y + cropWidth / 2)
+      ..translate(data.x + cropWidth / 2, data.y + cropHeight/2)
```
Comment (vanlooverenkoen) [inline lib/src/widgets/custom_image_crop_widget.dart:239]: format

File: lib/src/widgets/custom_image_crop_widget.dart [hunk 10]
```diff
@@ -188,45 +194,69 @@ class _CustomImageCropState extends State<CustomImageCrop> with CustomImageCropL
...
       ..style = PaintingStyle.fill;
-    canvas.drawRect(Rect.fromLTWH(0, 0, cropWidth, cropWidth), bgPaint);
+    canvas.drawRect(Rect.fromLTWH(0, 0, cropWidth, cropHeight), bgPaint);
     canvas.save();
     canvas.clipPath(clipPath);
     canvas.transform(matrix4Image.storage);
     canvas.drawImage(imageAsUIImage!, Offset(-imageAsUIImage!.width / 2, -imageAsUIImage!.height / 2), imagePaint);
     canvas.restore();
-
+
```
Comment (vanlooverenkoen) [inline lib/src/widgets/custom_image_crop_widget.dart:252]: remove unwanted spacing

**Review by vanlooverenkoen — CHANGES_REQUESTED** [review]

File: lib/src/widgets/custom_image_crop_widget.dart (hunk 8 above)
Comment (demonskp) [reply lib/src/widgets/custom_image_crop_widget.dart]: I think it is inappropriate to use if else in switch, so I chose another way.

**Comment by demonskp** [comment]
review

**Comment by NicolaVerbeeck** [comment]
There still seem to be formatting issues. Did you run dart format on the dir?

**Comment by demonskp** [comment]
> There still seem to be formatting issues. Did you run dart format on the dir?
When I ran dart format, the format of the old code was also changed. It seems that our formatting rules are inconsistent.

**Comment by vanlooverenkoen** [comment]
I have added travis with extra pr checks. Can you fix the conflicts & trigger another build (new commit)

**Comment by demonskp** [comment]
> I have added travis with extra pr checks. Can you fix the conflicts & trigger another build (new commit)

I'm sorry, but I recently changed jobs. So no time to fix these issues, I'll fix it later.

**Comment by vanlooverenkoen** [comment]
Any progress on this pr? @demonskp

**Comment by vanlooverenkoen** [comment]
@demonskp any progress on this pr?

**Comment by PavieOlivier** [comment]
Any update @demonskp ??

**Pull request closed without merging**